sqlglot/dialects/tsql.py

```diff
@@ -667,7 +667,19 @@ def create_sql(self, expression: exp.Create) -> str:
                 if kind == "SCHEMA":
                     sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.schemata WHERE schema_name = {identifier}) EXEC('{sql}')"""
                 elif kind == "TABLE":
-                    sql = f"""IF NOT EXISTS (SELECT * FROM information_schema.tables WHERE table_name = {identifier}) EXEC('{sql}')"""
+                    assert table
```

Comment:
Shouldn't we also apply this fix to the other cases for `kind`?

Reply:
it's selecting from `sys.indexes` using an object id, and that table doesn't include schema/catalog names

Reply:
sorry, that was for index -- i'll add it for schema

Reply:
and schema can only come in unqualified, so identifier will always work

Reply:
i see, makes sense